scripts/ui.ts

```diff
@@ -494,6 +465,61 @@ async function setupAndRunDisplay<T>(displayUrl: string, methodName: string, ...
     });
 }
 
+/**
+ * Resolves when portal login on Main MySky completes successfully.
+ *
+ * We register a storage event listener inside a promise that resolves the
+ * promise when we detect a successful portal login. The successful login is
+ * signaled via local storage. Any value other than "1" is considered to be an
+ * error message. If a successful login is not detected within a given timeout,
+ * then we reject the promise.
+ *
+ * @returns - An empty promise.
+ */
+async function resolveOnMySkyPortalLogin(): Promise<void> {
+  log("Entered resolveOnMySkyPortalLogin");
+
+  const abortController = new AbortController();
+
+  // Set up a promise that succeeds on successful login in main MySky, and fails
+  // when the login attempt returns an error.
+  const promise1 = new Promise<void>((resolve, reject) => {
+    const handleEvent = async ({ key, newValue }: StorageEvent) => {
+      if (key !== PORTAL_LOGIN_COMPLETE_SENTINEL_KEY) {
+        return;
```

Comment:
so this makes it so `promise1` never resolves, are we good with that?

Reply:
Yeah, we only want the promise to resolve when the right storage key is encountered. Any other storage key shouldn't trigger a `resolve`.

Reply:
Should we `reject` here, then?

Reply:
No! 😅 We don't want login to fail if a different key is set for whatever reason. If there is another storage event listener for that key, it will also be triggered. Here, we are only interested in the specific key we are listening for.